hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/DataConnectionResolver.java

```diff
@@ -110,20 +107,31 @@ public static List<DataConnectionCatalogEntry> getAllDataConnectionEntries(
         List<DataConnectionCatalogEntry> dataConnections =
                 dataConnectionService.getConfigCreatedDataConnections()
                                      .stream()
-                                     .map(dl -> new DataConnectionCatalogEntry(
-                                             dl.getName(),
-                                             dataConnectionService.typeForDataConnection(dl.getName()),
-                                             dl.getConfig().isShared(),
-                                             dl.options(),
+                                     .map(dc -> new DataConnectionCatalogEntry(
+                                             dc.getName(),
+                                             dataConnectionService.typeForDataConnection(dc.getName()),
+                                             dc.getConfig().isShared(),
+                                             dc.options(),
                                              DataConnectionSource.CONFIG))
-                                     .collect(Collectors.toList());
+                                     .collect(toList());
 
         // And supplement them with data connections from sql catalog.
         // Note: __sql.catalog is the only source of truth for SQL-originated data connections.
         dataConnections.addAll(dataConnectionStorage.dataConnections());
         return dataConnections;
     }
 
+    public static List<Object[]> getAllDataConnectionNameWithTypes(
+            DataConnectionServiceImpl dataConnectionService) {
+        List<DataConnection> conn = new ArrayList<>();
+        conn.addAll(dataConnectionService.getConfigCreatedDataConnections());
+        conn.addAll(dataConnectionService.getSqlCreatedDataConnections());
+
+        return conn.stream()
+                   .map(dc -> new Object[] {dc.getName(), new ArrayList<>(dc.resourceTypes())})
```

Comment:
Here is the array list in Object[] result. I don't think we support that.

Reply:
https://github.com/hazelcast/hazelcast/pull/24356/commits/b4df8f0363a24a35130ecd89d697105fb5413292